### ISSUE TITLE
Upgrading from 1.2.9 Fedora install

### ISSUE THREAD

**Issue by pvPrice** [craigk5n/webcalendar#257]
Actions:
Upgrade Fedora WebCalendar install from 1.2.9 to 1.3.0;
next, get WebCalendar 1.3.0 to function with php8

1 - Replaced WebCalendar with webcalendar-master.zip download (1.3.0)

Database changes required:
1 - webcal_config
	add cal_setting = WEBCAL_PROGRAM_VERSION; cal_value = V1.3.0
	
	note:  without this, the calendar is always in install mode
	
2 - webcal_user
	modify cal_enabled = Y
	
	note:  	my 1.2.9 set 2 calendars to N
			I was never able to log in
			The error is always Invalid Login, until this is Y

Fatal errors:
1)	./includes/functions.php - L4864
	uncaught type error abs()
	$catNum = abs( (int) $event->getCategory() );
	// add cast (int)
	
2)	./includes/gradient.php - L138
	array & string offset access sytax with curly brackets no longer supported
	if ( $path [ strlen ( $path) - 1 ] == '/'
	// replace {} with []
	
	./security_audit.php - L206
	if ( $path [ strlen ( $path ) - 1 ] == '/'
	// replace {} with []

3)	./edit_entry_handler.php L922
	call to undefined function each()
	current
	while ( list ( $old_participant, $dummy ) = each ( $old_status ) ) {
	replaced with
	foreach( $old_status as $old_participant => $dummy) {
	$$old_participant = $dummy;
	
	note:  	This construct eliminates the fatal error, but I'm not sure if it is correct.
			I don't use the calendar with emails.
	note:	There are a lot of each() functions; I'm not sure what to do with them.
			Nothing is flagging an error for me, but I'm not using all it's functionality.
	
Deprecated errors:
A)  Optional parameters must follow after all the required parameters

1)	get_byday (./includes/functions.php)
	Swap $date and $type parameters (#3 & #4)
	./includes/functions.php - 6 instances

2)	build_entry_popup (./includes/functions.php)
	Swap $time and $description parameters (#3 & #4)
	./docs/WebCalendar-Functions.html - 1 instance
	./includes/functions.php - 6 instances
	./list_unapproved.php - 1 instance
	./upcoming.php - 1 instance
	./week_details.php - 1 instance
	
3)	access_user_calendar (./access.php)
	Swap $other_user and $cal_can_xxx parameters (#1 & #2)
	./approve_entry.php - 2 instances
	./combo.php - 1 instance
	./day.php - 1 instance
	./del_entry.php - 4 instances
	./docadd.php - 1 instance
	./edit_entry.php - 2 instances
	./edit_entry_handler.php - 3 instances
	./events_ajax.php - 3 instances
	./access.php - 1 instance
	./includes/functions.php - 12 instances
	./includes/menu/index.php - 1 instance
	./includes/views.php - 1 instance
	./list_unapproved.php - 3 instances
	./month.php - 1 instance
	./reject_entry.php - 2 instances
	./rss_unapproved.php - 1 instance
	./search_handler.php - 1 instance
	./view_entry.php - 9 instances
	./view_t.php - 2 instances
	./week.php - 1 instance
	./ws/event_mod.php - 1 instance
	./ws/get_unapproved.php - 1 instance
	./year.php - 1 instance
	
4)	jscMenu_item (./includes/menu/index.php)
	Swap $url and $title (#2 & #3)
	./includes/menu/index.php - 43 instances

I hope this helps someone...
